ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/SupportsMempool.hs

```diff
@@ -79,6 +80,16 @@ data WhetherToIntervene
     -- them for it.
     deriving (Show)
 
+-- | Whether to keep track of the diffs produced by applying the transactions.
+--
+-- When getting a mempool snapshot, we will revalidate all the
+-- transactions but we won't do anything useful with the resulting
+-- state. We can safely omit computing the differences in this case.
```

Comment:
Please amend this comment to also discuss:

- This is a worthwhile optimization (and has been measured as such), since snapshotting the mempool is on the critical path of block minting.
- Eventually, the UTxO HD plan has always been for the ledger rules to construct the differences, instead of the Consensus layer computing them retroactively via `calculateDifferences`. Once that's true, we hope this optimization will no longer be worthwhile. That's part of the reason that we're content to just use a boolean isomorph instead and and yield empty `Diff`s instead of making it a GADT that makes the precise codomain (or just having two different functions, one with `Diff`s in the codomain and one without).